uwblueprint/paramedics-web
# Integrate SAML

## Overview
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

[Notion task](https://www.notion.so/uwblueprintexecs/Authentication-User-Setup-SAML-93cf50133f394109b7e6929ada1f3089)


## Changes
- Introduce Express again
- GraphiQL is back at localhost:4000/graphiql (bye bye to cool UI 😢)
- Add SAML certs

## Testing
- Requires a third party test identity provider (IdP), steps to set up below:
    -  run `docker pull kristophjunge/test-saml-idp` to fetch IdP container
    - run the below snippet to start the container with the right fields 
```
docker run --name=testsamlidp_idp \
-p 8080:8080 \
-p 8443:8443 \
-e SIMPLESAMLPHP_SP_ENTITY_ID=rowp-saml \
-e SIMPLESAMLPHP_SP_ASSERTION_CONSUMER_SERVICE=http://localhost:4000/login/callback \
-d kristophjunge/test-saml-idp
```
- go to `localhost:4000/login`, should see the below screen:
![image](https://user-images.githubusercontent.com/22237453/88486556-7f64d480-cf4c-11ea-930d-15c31e9873e7.png)
- This test IdP provides us with a sample user, with username `user1` and password `user1pass`

## To-do:
[ ] Move certs to environment variable (can it be static?)
- test with paramedics metadata (connecting to paramedics IdP)
- what logic can we move to the front end?
     - session management (what's needed in the front end/back end? whats the flow like?)
- Merge with other auth pr

Note: The IdP must be replaced in prod with a real IdP (Okta, Duo, etc.)
